src/test_l3_dsFPD.c

```diff
@@ -0,0 +1,716 @@
+    {"dsERR_OPERATION_FAILED", (int32_t)dsERR_OPERATION_FAILED},
+    {"dsErr_MAX", (int32_t)dsErr_MAX},
+    {NULL, -1}
+};
+
+const static ut_control_keyStringMapping_t dsFrontPanelIndicatorTable[] = {
+    {"dsFPD_INDICATOR_MESSAGE", (int32_t)dsFPD_INDICATOR_MESSAGE},
+    {"dsFPD_INDICATOR_POWER", (int32_t)dsFPD_INDICATOR_POWER},
+    {"dsFPD_INDICATOR_RECORD", (int32_t)dsFPD_INDICATOR_RECORD},
+    {"dsFPD_INDICATOR_REMOTE", (int32_t)dsFPD_INDICATOR_REMOTE},
+    {"dsFPD_INDICATOR_RFBYPASS", (int32_t)dsFPD_INDICATOR_RFBYPASS},
+    {"dsFPD_INDICATOR_MAX", (int32_t)dsFPD_INDICATOR_MAX},
+    {NULL, -1}
+};
+
+const static ut_control_keyStringMapping_t dsFrontPanelStateTable[] = {
+    {"dsFPD_STATE_OFF", (int32_t)dsFPD_STATE_OFF},
+    {"dsFPD_STATE_ON", (int32_t)dsFPD_STATE_ON},
+    {"dsFPD_STATE_MAX", (int32_t)dsFPD_STATE_MAX},
+    {NULL, -1}
+};
+const static ut_control_keyStringMapping_t dsFrontPanelColorTable[] = {
+    {"dsFPD_COLOR_BLUE", (int32_t)(0x0000FF)},
+    {"dsFPD_COLOR_GREEN", (int32_t)(0x00FF00)},
+    {"dsFPD_COLOR_RED", (int32_t)(0xFF0000)},
+    {"dsFPD_COLOR_YELLOW", (int32_t)(0xFFFFE0)},
+    {"dsFPD_COLOR_ORANGE", (int32_t)(0xFF8C00)},
+    {"dsFPD_COLOR_WHITE", (int32_t)(0xFFFFFF)},
+    {NULL, -1}
+};
+const static ut_control_keyStringMapping_t dsFrontPanelLEDState[] = {
+    {"dsFPD_LED_DEVICE_NONE", (int32_t)dsFPD_LED_DEVICE_NONE},
+    {"dsFPD_LED_DEVICE_ACTIVE", (int32_t)dsFPD_LED_DEVICE_ACTIVE},
+    {"dsFPD_LED_DEVICE_STANDBY", (int32_t)dsFPD_LED_DEVICE_STANDBY},
+    {"dsFPD_LED_DEVICE_WPS_CONNECTING", (int32_t)dsFPD_LED_DEVICE_WPS_CONNECTING},
+    {"dsFPD_LED_DEVICE_WPS_CONNECTED", (int32_t)dsFPD_LED_DEVICE_WPS_CONNECTED},
+    {"dsFPD_LED_DEVICE_WPS_ERROR", (int32_t)dsFPD_LED_DEVICE_WPS_ERROR},
+    {"dsFPD_LED_DEVICE_FACTORY_RESET", (int32_t)dsFPD_LED_DEVICE_FACTORY_RESET},
+    {"dsFPD_LED_DEVICE_USB_UPGRADE", (int32_t)dsFPD_LED_DEVICE_USB_UPGRADE},
+    {"dsFPD_LED_DEVICE_SOFTWARE_DOWNLOAD_ERROR", (int32_t)dsFPD_LED_DEVICE_SOFTWARE_DOWNLOAD_ERROR},
+    {"dsFPD_LED_DEVICE_MAX", (int32_t)dsFPD_LED_DEVICE_MAX},
+    {NULL, -1}
+};
+
+void readAndDiscardRestOfLine(FILE* in)
+{
+    int c;
+    while ( (c = fgetc(in)) != EOF && c != '\n');
+}
+
+/**
+* @brief Initialization of the Device settings - Front Panel Module
+*
+* This test provides a scope to Intialize the Front Panel module.
+
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+
+void test_1_dsFPD_hal_Init(void)
+{
+   gTestID = 1;
+   dsError_t status = dsERR_NONE;
+
+   UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+   // Step 1: Call dsFPInit()
+   UT_LOG_INFO("Calling dsFPInit()");
+   status = dsFPInit();
+   UT_LOG_INFO("Result dsFPInit(OUT:dsError_t:[%s])",UT_Control_GetMapString(dsFrontPanelErrorCodeTable, status));
```

Comment:
this is not a output parameter of Init so make like "Result dsFPInit() dsError_t:[%s]"

Reply:
updated